### PR TITLE
replace font popup menu with NSFontPanel

### LINKED ISSUE #781: font selector in v0.32.0 doesn't find all monospace fonts
i have a couple of different Inconsolata fonts installed (Inconsolata Nerd Font from https://www.nerdfonts.com/, Inconsolata for Powerline from https://github.com/powerline/fonts/tree/master/Inconsolata), and the selector in v0.32.0's preferences dialog doesn't detect any of them.
Screen Shot 2020-03-05 at 11 58 28 AM
is there additional information i can provide that would be helpful in debugging?

### PR DESCRIPTION
**Pull request by apaleslimghost**
replaces the font and size popups with a button:

Screenshot 2020-04-20 at 22 16 18

which opens this familiar thing:

Screenshot 2020-04-20 at 22 17 08

notably, this means selecting font weights & styles other than the font's default are now possible.

i can't find a way to restrict the selector to just the `Fixed Width` system collection. it's possible to disable the extraneous font style buttons on the toolbar, but if the panel is visible and you switch preference panes, or from the preferences window to another VimR window, because the other views/windows don't implement `validModesForFontPanel` the panel changes around and it looks a bit weird.

probably closes #781.

### CODE REVIEW

File: VimR/VimR/AppearancePref.swift [hunk 1]
```diff
@@ -294,8 +273,8 @@ class AppearancePref: PrefPane, NSComboBoxDelegate, NSControlTextEditingDelegate
   }
 
   private func updateViews() {
-    if let familyName = self.font.familyName { self.fontPopup.selectItem(withTitle: familyName) }
-    self.sizeCombo.stringValue = String(Int(self.font.pointSize))
+    sharedFontPanel.setPanelFont(font, isMultiple: false)
+    self.fontPanelButton.title = font.displayName.map { "\($0) \(font.pointSize)" } ?? "Show fonts..."
```
Comment (apaleslimghost) [inline VimR/VimR/AppearancePref.swift:277]: is the styling, text etc of this button acceptable?

Screenshot 2020-04-20 at 22 26 46

**Comment by apaleslimghost** [comment]
> i can't find a way to restrict the selector to just the Fixed Width system collection

[this is possible with a subclass of `NSFontManager`](https://stackoverflow.com/questions/18011497/filtering-fonts-from-nsfontpanel/18012386#18012386).

**Comment by hakamadare** [comment]
@qvacua do you have thoughts about this?  i'd love to see this fix merged

**Comment by gpmcadam** [comment]
> > i can't find a way to restrict the selector to just the Fixed Width system collection
> 
> [this is possible with a subclass of `NSFontManager`](https://stackoverflow.com/questions/18011497/filtering-fonts-from-nsfontpanel/18012386#18012386).

Is there a good enough reason to limit the selection? If someone wants to use `Comic Sans MS` then let them select it! 😅

**Comment by apaleslimghost** [comment]
the issue with using a non-monospace font is that VimR still draws it with even spacing, and it looks awful

**Comment by kuntau** [comment]
I need this the most.... Can you share the compiled binary @apaleslimghost ?

**Comment by gpmcadam** [comment]
I'm starting to think this project is dead. Is anyone aware of an active fork?

**Comment by qvacua** [comment]
I'm sorry for late reply. I am going to review it timely.

**Comment by qvacua** [comment]
@gpmcadam : I just had may things to do such that I did not have time to do things for VimR.

File: VimR/VimR/AppearancePref.swift (hunk 1 above)
Comment (qvacua) [reply VimR/VimR/AppearancePref.swift:277]: 👍

**Comment by qvacua** [comment]
> the issue with using a non-monospace font is that VimR still draws it with even spacing, and it looks awful

Yes, I made extra effort to render it in a grid-like structure since that's the way (Neo)vim assumes how the text is displayed. Since I also don't know how to limit the selection in the font panel, one could warn the user when they select a variable width font.

**Comment by gpmcadam** [comment]
> @gpmcadam : I just had may things to do such that I did not have time to do things for VimR.

No problem, good to hear you're back. Looking forward to more releases! 🎉 Thanks

**Pull request merged**